player/sub.c

```diff
@@ -167,8 +168,16 @@ static bool init_subdec(struct MPContext *mpctx, struct track *track)
     if (!track->demuxer || !track->stream)
         return false;
 
+    char *force_configdir = mpctx->global->configdir;
+    mpctx->global->configdir =
+            bstrto0(mpctx->global, mp_dirname(mpctx->filename));
```

Comment:
This is very, very wrong. Assigning this field affects all threads and cannot be done safely.

Reply:
Thanks for your review. Initially, I also would like to avoid changing configuration directory, but the video file name could not be gotten in the setting font folder function, i.e. https://github.com/mpv-player/mpv/blob/dcc3c2eb3877f9876cc218b1e5f1187b81036c49/sub/ass_mp.c#L142 The nearest location where I can fetch video file name is the location where I change. Indeed, I worried that assigning `mpctx->global->configdir` is likely to affect all other things. I attempted to monitor when and where this field is accessed. The result is that my modification seems to affect only one situation when subtitle renderer starts to load fonts, i.e. `fonts` folder, `fonts.conf` and `subfont.ttf`. Nevertheless, I still fear that my PR will cause problems since I am afraid to overlook some situations.